elfi/distributions.py

```diff
@@ -32,6 +32,10 @@ def pdf(self, x, *params, **kwargs):
     def logpdf(self, x, *params, **kwargs):
         raise NotImplementedError
 
+    @property
+    def name(self):
+        return self._name or Distribution.__class__.__name__
```

Comment:
Now the latter would always output just `Distribution`. We either need to remove the signatures from the methods or create another abstract class for static classes. I'd prefer the duck typing approach, but you can decide.